.github/workflows/deploy-website.yml

```diff
@@ -0,0 +1,86 @@
+name: Deploy website to Pages
+
+on:
+  push: # Runs on pushes targeting the default branch
+    branches:
+      - main
+    paths:
+      - 'docs/**'
+  workflow_dispatch: # Allows you to run this workflow manually from the Actions tab
+
+permissions: # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency: # Allow one concurrent deployment
+  group: "pages"
+  cancel-in-progress: true
+
+defaults: # Default to bash
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Detect package manager
+        id: detect-package-manager
+        working-directory: ./docs
+        run: |
+          if [ -f "yarn.lock" ]; then
+            echo "manager=yarn" >> $GITHUB_OUTPUT
+            echo "command=install" >> $GITHUB_OUTPUT
+            exit 0
+          elif [ -f "package.json" ]; then
+            echo "manager=npm" >> $GITHUB_OUTPUT
+            echo "command=ci" >> $GITHUB_OUTPUT
+            exit 0
+          else
+            echo "Unable to determine packager manager"
+            exit 1
+          fi
```

Comment:
is there any specific benefit in this dynamic detection? could we maybe just assume it's going to be yarn?